### DISCUSSION TITLE
1.2.0 is a big deal!

### DISCUSSION THREAD

**Discussion by geowarin** [diasurgical/DevilutionX#1534] [Show and tell]
I saw a comment by @AJenbo in a PR stating that 1.2.0 attracted new developers.

As far as I am concerned, this is totally true. Replaying the title I loved when I was a kid, with a gamepad, many bugfixes and and  all, is a delight.

As a developper, looking at the actual codebase and seeing it evolve and being refactored is really fascinating.

You guys are doing an awesome work! 
The thing I think you could do better is giving some guidelines on the overall direction of the project so that other people can make modifications that are relevant.

Anyways, as a lurker, I thought I should drop by and say thank you for the very hard work you guys have been doing for years.
:heart:

**Comment by AJenbo** [comment]
Thanks.

A lot of things are still being figured out and up in the air. I have to define and go over clang-tidy and move files, header, includes and definitions around a bit, while others are investigating different way to clean up different parts of the code.

One of the most helpful think atm would be adding tests and documentation as that helps us avoid introducing bugs as we refactor.

**Reply by geowarin** [reply]
Indeed, I figured as much. 
I began writing some unit tests but since the code base is using a lot of global variables, this becomes messy very quickly. 
One thing I've been pondering is the usage of mock vs using `#ifndef` to remove some problematic code.
Usage of `LoadFileInMem`, or other methods that load assets, is problematic.

Any advice on what and how the code should be tested?

**Reply by AJenbo** [reply]
we could use the share ware in some tests, the code should be as little aware that it is running tests as possible.

**Reply by AJenbo** [reply]
the other common task that the code needs is removing duplication. splitting up large functions. flatening nestin. Not declaring variables before first initialization.

**Comment by dcworldwide** [comment]
I also really enjoy watching the project evolve. This is one of my fav hobbies away from work. Thankyou to all the talented devs involved
